src/main/java/io/technical_tasks/spring_web_app_for_payment_processing/service/PaymentService.java

```diff
@@ -0,0 +1,99 @@
+package io.technical_tasks.spring_web_app_for_payment_processing.service;
+
+import io.technical_tasks.spring_web_app_for_payment_processing.payment.Payment;
+import io.technical_tasks.spring_web_app_for_payment_processing.payment.PaymentRequest;
+import io.technical_tasks.spring_web_app_for_payment_processing.repository.PaymentRepository;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class PaymentService {
+
+    @Autowired
+    private PaymentRepository paymentRepository;
+
+    @Autowired
+    private GeoIPService geoIPService;
+
+    @Transactional
+    public Payment createPayment(PaymentRequest paymentRequest, HttpServletRequest request) {
+        String ip = request.getHeader("X-Forwarded-For");
+        if (ip == null || ip.isEmpty()) {
+            ip = request.getRemoteAddr();
+        }
+
+        System.out.println("Using IP address: " + ip);
+
+        String countryCode = geoIPService.getCountryCode(ip);
+
+        Payment payment = new Payment();
+        payment.setAmount(paymentRequest.getAmount());
+        payment.setDebtorIban(paymentRequest.getDebtorIban());
+        payment.setTimeStamp(LocalDateTime.now());
+        payment.setCountryCode(countryCode);
+
+        return paymentRepository.save(payment);
+    }
+
+    public List<Payment> getPayments(String debtorIban) {
+        if (debtorIban == null) {
+            return paymentRepository.findAll();
+        } else {
+            return paymentRepository.findByDebtorIban(debtorIban);
+        }
+    }
+
+    @Transactional
+    public String readCSVPaymentFile(MultipartFile file, HttpServletRequest request) {
+        List<String> errors = new ArrayList<>();
+        PaymentRequest paymentRequest = new PaymentRequest();
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(file.getInputStream()))) {
+            String line;
+            boolean isFirstLine = true;
+
+            while ((line = reader.readLine()) != null) {
+                if (isFirstLine) {
+                    isFirstLine = false;
+                    continue;
+                }
+
+                String[] fields = line.split(",");
```

Comment:
Es laikam ieteiktu labāk paņemt kādu bibliotēku, kas apstrādā CSV nevis pašam to darīt manuāli. Var paskatīties šeit:
https://www.baeldung.com/java-csv-file-array tur ir par OpenCSV, vēl viens variants ir šis:
https://www.baeldung.com/apache-commons-csv